javalib/src/main/scala/java/util/concurrent/atomic/AtomicReferenceArray.scala

```diff
@@ -0,0 +1,41 @@
+package java.util.concurrent.atomic
+
+class AtomicReferenceArray[E <: AnyRef](
+    length: Int) extends Serializable {
+
+  def this(array: Array[E]) = {
+    this(array.size)
+    System.arraycopy(array, 0, inner, 0, length)
+  }
+
+  private val inner: Array[E] = (new Array(length)).asInstanceOf[Array[E]]
```

Comment:
That's not a valid cast.
If you don't want the `js.Array` solution, then you need to create a `new Array[AnyRef]`, type `inner` as `Array[AnyRef]`, and cast the result in `get(i)`.